src/ds/ptrwrap.h

```diff
@@ -101,14 +101,16 @@ namespace snmalloc
   template<typename T, capptr_bounds bounds>
   struct CapPtr
   {
-    T* unsafe_capptr;
+    uintptr_t unsafe_capptr;
 
     /**
      * nullptr is implicitly constructable at any bounds type
      */
-    CapPtr(const std::nullptr_t n) : unsafe_capptr(n) {}
+    constexpr SNMALLOC_FAST_PATH CapPtr(const std::nullptr_t n)
+    : unsafe_capptr(reinterpret_cast<uintptr_t>(n))
```

Comment:
I'm surprised this reinterpret_cast is allowed in a constexpr.

Reply:
So am I... oh.

This is C++ templating biting us once again: that `reinterpret_cast` isn't actually instantiated as a `constexpr` anywhere and so nobody notices that there's a spec violation lurking in the template AST.  Greeeeeat.

But that answers your other question: no, `pointer_offset` cannot be usefully made `constexpr` (though you'll get no errors if you do, it just won't be useful).